### DISCUSSION TITLE
Visualisierung und Datenspeicherung abseits von Home Assistant

### DISCUSSION THREAD

**Discussion by rs3hc** [hoylabs/OpenDTU-OnBattery#750] [General]
Guten Morgen liebe Freunde der Sonne. 😁

Gibt es abseits von Home Assistant und Co eine kleine Plattform zur Speicherung und Visualisierung der Daten  von OpenDTU-OnBattery?

**Comment by spcqike** [comment]
Bei mir lungern die üblichen verdächtigen im Netzwerk herum: Node-Red, influxDB und Grafana. 

An einem Remote Standort habe ich eine raspberry1 auf dem der Volkszähler läuft. 

Sind aber beides weniger out of the box Plattformen.

**Comment by schlimmchen** [comment]
Hast du mal an Prometheus und Grafana gedacht? Die Unterstützung für Prometheus ist im -OnBattery-Teil leider lückenhaft, siehe z.B. #717, aber es wäre ein Anfang. Ich denke allerdings auch über einen MQTT Exporter für Prometheus nach. Weil der MQTT Broker mehr oder weniger die vollständige Sicht der Dinge hat, sollten dessen Daten "einfach" in eine time-relational database transportiert werden, worauf man dann beliebige Auswertungen machen kann. Ich hab da mal reingeschnuppert, aber mich ablenken lassen und es ist nichts fertig, daher kann ich ehrlich gesagt auch nur darauf hinweisen, dass es das gibt.

**Reply by rs3hc** [reply]
Ok das muss ich mir mal anschauen.

Bei mir ist das kein Problem, da ich eh HA laufen habe.

Aber mein Nachbar hat nix dergleichen und jedes Watt ist eins zu viel 🤣

Ist evtl ein kostenloser Clouddienst bekannt der eine DB mit grafischer auswertung bietet?
Ist ja jetzt nichts sicherheitsrelevantes... 

LG aus dem Erzgebirge 👍

**Reply by gitisgreat2023** [reply]
Ich nutze für Datenauswertung iobroker auf einem tv-box. Das sind kleine system-on-a-chip Systeme die alles dabei haben, inkl. powersupply. Kostet bei Ali 50€, für ein H96 Max S905X3 mit 4GB RAM und 128GB EMMC. Armbian drauf und läuft. Nimmt so 2-4 Watt max (headless, ich nutze es mit LAN, WiFi geht ja auch). Ist tatsächlich super sparsam. Bei mir hängt auch noch einen Zigbee hub dran, um den Gasmeter (Fenstordetektor sensor trick, 10€ Lösung) auszulesen und andere Zigbee sensoren.
iobroker reicht für einfache Auswertungen. Ich habe cron ähnlich scripte, die machen hourly, daily, weekly, etc Datensnapshots die ich dann visualisieren kann. Nur der Datenexport ist schlecht, für einige Sachen nutze ich Node-Red um die Daten direkt in einem csv zu schreiben.

**Reply by andreasonline** [reply]
Es gibt einen öffentlichen Volkszähler auf dem sich jeder auch private Kanäle anlegen kann. Durch die lange kryptische UUID kann nur der die Daten lesen, der die ID kennt und ist somit auch recht Sicher. Ich habe mir einen eigenen Volkszähler bei meinem Hoster auf meiner Domäne installiert. weitere infos zum Volkszähler unter https://volkszaehler.org/
Mein Tasmota Power-Meter sendet schon seine Daten dort hin ( siehe https://github.com/helgeerbe/OpenDTU-OnBattery/discussions/750#discussioncomment-9166360) , für die openDTU on Battery suche ich noch eine Lösung.

**Comment by andreasonline** [comment]
Hallo, ich würde gerne den Volkszähler direkt zur grafischen Virtualieirung nutzen. Dazu müsste der ESP nur im intervall von 5 Sekunden Werte, die im Live-View angezeigt werden per HttpPost an den Volkszähler senden. 
In den Tasmota Power-Meter habe ich das über das e-Script ganz einfach einfügen können. Somit erfasse ich schon damit meinen aktuellen Verbrauch.
![Tasmota e-script](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/166002171/0a8c98c9-5d55-433e-8165-822689de2bf1)
![Volkszähler Anzeige](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/166002171/a36ee955-6022-4a5f-81c3-bc7635665dbe)
![Volkszähler Stunden summiert](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/166002171/825c7dfb-2cc2-49da-8138-3d844f4c5049)

Das Format sieht wie fogt aus:
```
http://<Volkszähler Server>/middleware.php/data/<uuid>.json?&operation=add&value=<Wert>
<Volkszähler Server> = IP oder DNS des Volkszähler
<uuid> = ID des Kanals auf dem Volkszähler bsp. ae53c580-1234-5678-90ab-cdefghijklmn für jeden zu erfassenden Wert z.B. MPPT Gesammtleistung wird auf dem Volkszähler einen eigenen Kanal/ID erzeugt.
<Wert> = zu übertragender Wert z.B. MPPT Gesammtleistung die im Live-View angezeigt wird.
``` 
Für mich wäre es ausreichend die MPPT Gesammtleistung und die Inverter Gesammtleistung auf diese Weise zum Volkszähler zu senden. Im prinzip könnte man natürlich jeden Wert auf diese weise senden.
Sinnvoll ist sicherlich ein intervall von 5 Sekunden zu verwenden, für den gesendete Wert wäre jedoch eine Mittelwert aus den letzten 5 Sekunden sinnvoll, da der Volkszähler die Möglichkeit hat Verbrauchswerte zu Summenverbrauch pro Stunde, Tag, Monat zusammenzufassen. 
Ich habe versucht dies selbst in den Code einzufügen, bin jedoch kein Progamierer und nicht weit genug den bestehenden Code zu verstehen. Habe bisher nur kleinere Projekte auf Arduino oder ESP geschrieben. Kann mir da jemand helfen?

**Comment by andreasonline** [comment]
Es gibt einen öffentlichen Volkszähler auf dem sich jeder auch private Kanäle anlegen kann. Durch die lange kryptische UUID kann nur der die Daten lesen, der die ID kennt und ist somit auch recht Sicher. Ich habe mir einen eigenen Volkszähler bei meinem Hoster auf meiner Domäne installiert. weitere infos zum Volkszähler unter https://volkszaehler.org/
Vieleicht gibt es ja noch mehrer Interessenten, die sich eine Erfassung der Leistung ohne IO-Brokerüber den Volkszähler wünschen. 
Ich habe zwar schon daran gedacht einen weiteren ESP zu verwenden, der die Daten per api von der openDTU on Battery ausließt und an der Volkszähler sendet, bin jedoch der Meinung, dass diese kleine Aufgabe auch durch die openDTU on Battery direkt erledigt werden könnte. Weniger Hardware = weniger Fehlerquellen.

**Reply by spcqike** [reply]
HTTP finde ich persönlich dort zu schwer gewichtig. 

Was spricht dagegen, auf den Server wo der Volkszähler läuft auch MQTT laufen zu lassen? MQTT ist für die Übertragung der Daten viel schneller und Ressourcen schonender und muss nicht erst einprogrammiert werden. 

Die Daten die du per MQTT bekommst kannst du dann per Script in deinen volkszähler schreiben. Das sollte schneller gehen und zu weniger Problemen auf dem esp führen.

**Reply by andreasonline** [reply]
Der Volkszähler läuft auf dem Hosting des Provider, nicht auf einem virtuellen Server. Gibt es ein MQTT Server den man auf einem Webhosting installieren kann?
